enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -374,3 +343,41 @@ Provide a recipe for customer's to deploy their own log collector to move the re
 
 ## Infrastructure Needed
 * Future target endpoints may require special infrastructure or licencing to properly test.  For example, Splunk is destination requested by many users but is a licensed product.  The free dev tier may not provide sufficient capacity to fully test
+* Scale and stress tests require intensive use of a large cluster for an extended period of time.
+## Future plans
+
+As well as serving the current GA requirements, the log forwarding API has been designed with the following future requirements in mind.
+
+### Stand-alone log forwarding. ###
+
+Deploy log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch) Forwarding will be a stand-alone system independent of any log store. This decoupling will let us test forwarding separately, and let customers to switch off our managed store entirely while still using a managed and supported forwarder.
+
+### As a team lead (tenant), I’d like to configure secure log forwarding to the tool of my team's choice, separate from global config. ###
+
+Introduce a namespace-scoped LogForwarder. The API is a restricted version of the ClusterLogForwarder API:
+- can't select on namespaces (only get logs from containers in own namespace)
+- can't use Infrastructure or Audit inputs
+
+Although there could be many `LogForwarder` objects, there is still only one collector. The operator would join all the configurations and compile them to a single collector configuration. It would also enforce the limitations of namespace-scoped forwarders.
```

Comment:
It may be paranoid thinking, and is predicated on having to restart collectors for effect config changes, but this creates a possibility for disruption of all collectors on the entire cluster if a tenant creates/modifies/destroys a namespace-scoped LogForwarder in a tight loop.

Reply:
You are not paranoid if they really are out to get you. My statement really applies to how we can layer the LF APIs over our existing solution. Once we have the API in place we will need to start looking at bottlenecks and excessive coupling, and possibly introduce intermediate processes to buffer/route/partition the system. Your example is a fine one, and we should measure it, and decide if we need to improve our design, before we release namespaced brokers.